### PR TITLE
remove old unnecessary / unused counter init code

### PR DESCRIPTION
**Pull request by danielnorberg**
# Hey, I just made a Pull Request!

## Description
remove old unnecessary / unused counter init code

## Motivation and Context
It is unnecessary and a little unsettling to mutate all the counter limits on startup.

## Checklist for PR author(s)

- [x] Changes are covered by unit test
- [x] All tests pass
- [x] Code coverage check passes
- [ ] Error handling is tested
- [ ] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] (optional) Changes are covered by system test
- [ ] Relevant documentation updated
- [x] This PR has NO breaking change to public API
- [ ] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [ ] This PR has been incorporated in release note for the coming version
- [ ] Risky changes introduced by this PR have been all considered

### CODE REVIEW

File: styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java [hunk 1]
```diff
@@ -457,48 +454,6 @@ void tickBackfillTriggerManager() {
...
-              });
-            } catch (IOException e) {
-              LOG.error("Error creating a counter limit for {}", resource, e);
-              throw new RuntimeException(e);
-            }
-          });
-      LOG.info("Finished initializing resources");
-
-      // Sync resources usage
-      if (storage.config().resourcesSyncEnabled()) {
```
Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java]: We should remove this flag related code as well.

Comment (honnix) [reply styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java]: And remove datastore configure after deploying.

Comment (danielnorberg) [reply styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java]: 👍

**Comment by honnix** [comment]
I will take the liberty to continue working on this. @danielnorberg

**Comment by danielnorberg** [comment]
👍

**Pull request merged**